src/network/serverpackethandler.cpp

```diff
@@ -287,9 +287,18 @@ void Server::handleCommand_Init2(NetworkPacket* pkt)
 	m_clients.event(pkt->getPeerId(), CSE_GotInit2);
 	u16 protocol_version = m_clients.getProtocolVersion(pkt->getPeerId());
 
-	std::string lang;
-	if (pkt->getSize() > 0)
-		*pkt >> lang;
+	// Receive the client locale
+	std::vector<std::string> locale;
+	u8 locale_count = 0;
+	try {
```

Comment:
i suggest a more interesting thing, learn NetworkPacket to serialize/deserialize vectors by adding operators << and >>

Reply:
Thanks for the suggestion. It is indeed a much better solution to have this part as a generic part of NetworkPacket.